dbt_platform_helper/commands/database_helpers.py

```diff
@@ -1,124 +1,205 @@
+import re
+from collections.abc import Callable
+from pathlib import Path
+
 import boto3
 import click
+from boto3 import Session
 
+from dbt_platform_helper.commands.environment import offline_command
+from dbt_platform_helper.commands.environment import online_command
+from dbt_platform_helper.constants import PLATFORM_CONFIG_FILE
+from dbt_platform_helper.exceptions import AWSException
+from dbt_platform_helper.utils.application import Application
+from dbt_platform_helper.utils.application import ApplicationNotFoundError
+from dbt_platform_helper.utils.application import load_application
 from dbt_platform_helper.utils.aws import Vpc
-from dbt_platform_helper.utils.aws import get_aws_session_or_abort
 from dbt_platform_helper.utils.aws import get_connection_string
 from dbt_platform_helper.utils.aws import get_vpc_info_by_name
+from dbt_platform_helper.utils.messages import abort_with_error
+from dbt_platform_helper.utils.validation import load_and_validate_platform_config
 
 
-def run_database_copy_task(
-    session: boto3.session.Session,
-    account_id: str,
-    app: str,
-    env: str,
-    database: str,
-    vpc_config: Vpc,
-    is_dump: bool,
-    db_connection_string: str,
-):
-    client = session.client("ecs")
-    action = "dump" if is_dump else "load"
-    response = client.run_task(
-        taskDefinition=f"arn:aws:ecs:eu-west-2:{account_id}:task-definition/{app}-{env}-{database}-{action}",
-        cluster=f"{app}-{env}",
-        capacityProviderStrategy=[
-            {"capacityProvider": "FARGATE", "weight": 1, "base": 0},
-        ],
-        networkConfiguration={
-            "awsvpcConfiguration": {
-                "subnets": vpc_config.subnets,
-                "securityGroups": vpc_config.security_groups,
-                "assignPublicIp": "DISABLED",
-            }
-        },
-        overrides={
-            "containerOverrides": [
-                {
-                    "name": f"{app}-{env}-{database}-{action}",
-                    "environment": [
-                        {"name": "DATA_COPY_OPERATION", "value": action.upper()},
-                        {"name": "DB_CONNECTION_STRING", "value": db_connection_string},
-                    ],
-                }
-            ]
-        },
-    )
-
-    return response.get("tasks", [{}])[0].get("taskArn")
+class MaintenancePageProvider:
+    def __init__(self):
+        self.offline = offline_command
+        self.online = online_command
 
 
 class DatabaseCopy:
```

Comment:
How do you feel about moving the classes etc. into a new `src` or similar directory and starting to tidily namespace it all?

Reply:
We've had similar thoughts. "commands" module will be where the thin click wrappers will live. "domain" module for the business logic and "providers" module for the common code that connects with things outside the domain logic (most of the "utils" module probably belongs in here)

Reply:
(I'll write this up in the confluence doc for people to comment on)